mypyc/build.py

```diff
@@ -283,14 +283,17 @@ def write_file(path: str, contents: str) -> None:
     want to write, skip writing so as to preserve the mtime
     and avoid triggering recompilation.
     """
+    # We encode it ourselves and open the files as binary to avoid windows
+    # newline translation
+    encoded_contents = contents.encode('utf-8')
     try:
-        with open(path, 'r', encoding='utf-8') as f:
-            old_contents = f.read()  # type: Optional[str]
+        with open(path, 'rb') as f:
+            old_contents = f.read()  # type: Optional[bytes]
     except IOError:
         old_contents = None
-    if old_contents != contents:
-        with open(path, 'w', encoding='utf-8') as f:
-            f.write(contents)
+    if old_contents != encoded_contents:
+        with open(path, 'wb') as f:
+            f.write(encoded_contents)
```

Comment:
I am not sure if this is relevant, but long time ago we had problems on Windows with incremental mode (see for example https://github.com/python/mypy/issues/3215). This is why we have some non-trivial logic in and around `FilesystemMetadataStore.write()`.  Maybe you can use the same here?